bin/check-efs-metric.rb

```diff
@@ -0,0 +1,142 @@
+  include Common
+  include CloudwatchCommon
+
+  option :efs_name,
+         description: 'Name of the EFS file system, matching the Name tag',
+         short: '-n VALUE',
+         long: '--name VALUE',
+         proc: proc(&:to_s),
+         required: true
+
+  option :metric_name,
+         description: 'CloudWatch metric in the AWS/EFS namespace: E.G. PercentIOLimit',
+         short: '-m VALUE',
+         long: '--metric VALUE',
+         proc: proc(&:to_s),
+         required: true
+
+  option :statistics,
+         description: 'Statistic to retrieve from CloudWatch: E.G. Minimum or Maximum',
+         short: '-s statistic',
+         long: '--statistic statistic',
+         default: 'Average',
+         proc: proc(&:to_s)
+
+  option :critical,
+         description: 'Return critical when the metric is at this VALUE, also see the --operator option',
+         short: '-c VALUE',
+         long: '--critical VALUE',
+         proc: proc(&:to_f),
+         required: true
+
+  option :warning,
+         description: 'Return warning when the metric is at this VALUE, also see the --operator option',
+         short: '-w VALUE',
+         long: '--warning VALUE',
+         proc: proc(&:to_f)
+
+  option :period,
+         description: 'CloudWatch metric statistics period, in seconds. Must be a multiple   of 60',
+         short: '-p VALUE',
+         long: '--period VALUE',
+         default: 60,
+         proc: proc(&:to_i)
+
+  option :compare,
+         description: 'Comparison operator for critical and warning thresholds: equal, not, greater, less',
+         short: '-o OPERATOR',
+         long: '--operator OPERATOR',
+         default: 'less'
+
+  option :unit,
+         description: 'CloudWatch metric unit, to be passed to the metrics request',
+         short: '-u UNIT',
+         long: '--unit UNIT'
+
+  option :no_data_ok,
+         description: 'Returns ok if no data is returned from the metric',
+         short: '-O',
+         long: '--allow-no-data',
+         boolean: true,
+         default: false
+
+  option :no_efs_ok,
+         description: 'Returns ok if the EFS file system specified by the -n option can not be found: This is useful if using a check in multiple environments where a file system may not always exist',
+         short: '-N',
+         long: '--allow-no-efs',
+         boolean: true,
+         default: false
+
+  option :aws_region,
+         description: 'AWS region',
+         short: '-r Region',
+         long: '--region REGION',
+         default: 'us-east-1'
+
+  # This is used by CloudwatchCommon to display the description for what is being checked.
+  def metric_desc
+    "#{config[:metric_name]} for #{config[:efs_name]}"
+  end
+
+  def run
+    config[:namespace] = 'AWS/EFS'
+    found_efs = false
+
+    efs = Aws::EFS::Client.new
+    filesystems = efs.describe_file_systems
+    filesystems.file_systems.each do |filesystem|
+      if filesystem.name.casecmp(config[:efs_name]).zero?
```

Comment:
I think `if filesystem.name.casecmp?(config[:efs_name])` is clearer in its intent and also has the added benefit of better handling of unicode characters (therefore more internationalization support). I had to go look at what the return output of `casecmp` was before I was positive I understood why it worked. Unfortunately this method was introduced in ruby 2.4 so maybe lets put a comment about switching it over when we drop `< 2.4` support at some point in the future. I'd also accept linking to the documentation for the method as well: https://ruby-doc.org/core-2.4.0/String.html#method-i-casecmp if it returns say `-1` that would non obvious to  most people without looking at the docs or source.